crd/external/podnetwork/api/v1alpha1/podnetwork.go

```diff
@@ -0,0 +1,71 @@
+//go:build !ignore_uncovered
+// +build !ignore_uncovered
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Important: Run "make" to regenerate code after modifying this file
+
+// +kubebuilder:object:root=true
+
+// PodNetwork is the Schema for the PodNetworks API
+// +kubebuilder:resource:scope=Namespaced
+// +kubebuilder:resource:shortName=pn
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="Status",type=string,priority=1,JSONPath=`.status.status`
+// +kubebuilder:printcolumn:name="Error Message",type=string,priority=1,JSONPath=`.status.errorMessage`
+// +kubebuilder:printcolumn:name="Address Prefixes",type=string,priority=1,JSONPath=`.status.addressPrefixes`
+// +kubebuilder:printcolumn:name="Network",type=string,priority=1,JSONPath=`.spec.vnetGuid`
+// +kubebuilder:printcolumn:name="Subnet",type=string,priority=1,JSONPath=`.spec.subnetResourceID`
+type PodNetwork struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   PodNetworkSpec   `json:"spec,omitempty"`
+	Status PodNetworkStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// PodNetworkList contains a list of PodNetwork
+type PodNetworkList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []PodNetwork `json:"items"`
+}
+
+// PodNetworkSpec defines the desired state of PodNetwork
+type PodNetworkSpec struct {
+	// +kubebuilder:validation:Optional
+	// customer vnet guid
+	VnetGUID string `json:"vnetGuid,omitempty"`
+	// customer subnet name
+	SubnetResourceID string `json:"subnetResourceID,omitempty"`
+}
+
+// Status indicates the status of PN
+// +kubebuilder:default=Nil
+// +kubebuilder:validation:Enum=Nil;Ready;InUse;SubnetNotDelegated
+type Status string
+
+const (
+	Nil                Status = "Nil"
```

Comment:
Do we need a `Nil` status? Conceptually, what is the difference between `"Nil"` and `""`?

Reply:
no, we don't need nil as an enum value.

Reply:
removed

Reply:
removed